validation-test/stdlib/FloatingPointConversion.swift.gyb

```diff
@@ -0,0 +1,192 @@
+    return "would be greater than"
+  }
+  return ""
+}
+
+var FloatingPointConversionTruncations = TestSuite("FloatingPointToFloatingPointConversionTruncations")
+var FloatingPointConversionFailures = TestSuite("FloatingPointToFloatingPointConversionFailures")
+
+% for Self, selfSignificandBits in floatNameToSignificandBits.iteritems():
+
+%  if Self == 'Float80':
+#if arch(i386) || arch(x86_64)
+%  end
+
+%  for OtherFloat, otherSignificandBits in floatNameToSignificandBits.iteritems():
+
+%   if OtherFloat == 'Float80':
+#if arch(i386) || arch(x86_64)
+%   end
+
+%   if otherSignificandBits <= selfSignificandBits:
+
+/// Always-safe conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we don't truncate floats that shouldn't be truncated.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  let result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectEqual(input, resultConvertedBack)
+}
+
+/// Never-truncating conversion from ${OtherFloat}.nextUp to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.1.0.nextUp") {
+  // Test that we don't truncate floats that shouldn't be truncated.
+  let input = (1.0 as ${OtherFloat}).nextUp
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectEqual(input, resultConvertedBack)
+}
+
+/// Never-nil failable conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}FailableConversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that nothing interesting happens and we end up with a non-nil, identical result.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(exactly: input)
+  expectNotEmpty(result)
+  var resultConvertedBack = ${OtherFloat}(result!)
+  expectEqual(input, resultConvertedBack)
+}
+
+/// Never-nil conversion from ${OtherFloat}.nextUp to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.1.0.nextUp") {
+  // Test that nothing interesting happens and we end up with a non-nil, identical result.
+  let input = (1.0 as ${OtherFloat}).nextUp
+  var result = ${Self}(exactly: input)
+  expectNotEmpty(result)
+  var resultConvertedBack = ${OtherFloat}(result!)
+  expectEqual(input, resultConvertedBack)
+}
+
+%   else:
+
+/// Always-truncating conversion from ${OtherFloat}.nextUp to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we truncate floats not representable by another type.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectNotEqual(input, resultConvertedBack)
```

Comment:
I think you can just check for infinity here, `expectEqual(${Self}.infinity, result)`.